horovod/common/operations.cc

```diff
@@ -1926,18 +1930,20 @@ void BackgroundThreadLoop(HorovodGlobalState& state) {
   // TODO: init.cu:645 WARN Cuda failure 'driver shutting down'
   //#if HAVE_NCCL
   //  for (auto it = horovod_global.streams.begin();
-  //       it != horovod_global.streams.end(); it++) {
+  //       it != horovod_global.streams.end(); ++it) {
   //    cudaStreamSynchronize(it->second);
   //  }
   //  for (auto it = horovod_global.nccl_comms.begin();
-  //       it != horovod_global.nccl_comms.end(); it++) {
+  //       it != horovod_global.nccl_comms.end(); ++it) {
   //    ncclCommDestroy(it->second);
   //  }
   //#endif
 
   // Notify all outstanding operations that Horovod has been shut down
   // and clear up the tensor table and message queue.
   std::vector<StatusCallback> callbacks;
+  // we know (# of elements put) == (size of tensor_table map).
+  callbacks.reserve(state.tensor_table.size()); 
```

Comment:
This happens only once, so let's revert this.

Reply:
done